### DISCUSSION TITLE
convertTokenView overriding existing User details

### DISCUSSION THREAD

**Discussion by DevilsAutumn** [wagnerdelima/drf-social-oauth2#168] [General]
Hi i'm trying to implement social authentication using google and github and i have an existing user model which has first and last name fields in it. 
- First i'm creating a normal user and then trying to connect that user account with a google account but as soon as i send a POST request on /auth/convert-token/ end-point with google access token with scope of email and user details. 
- This post request overrides my existing first name and last name with that i got from google details.
- Now if i connect my github account with the same user account it again overrides my first and last name.

I want that noneof my social accounts should override my existing user details, what is the best way to do so? 
Thankyou for the help!

**Comment by wagnerdelima** [comment]
Hi @DevilsAutumn,

This is the default behaviour of python-social-oauth. Your user name and family name will become the user name of the Google or Github account. If a name and user already exists, social-oauth will append a random number to it. 

I believe you can customise your code to not accept the changes to your username. But as far as the framework goes, it changes it automatically.

**Reply by DevilsAutumn** [reply]
Hi, thanks for the reply! I Just found out in the documentation of python social oauth  there is a [miscellaneous setting](https://python-social-auth.readthedocs.io/en/latest/configuration/settings.html#miscellaneous-settings) ```SOCIAL_AUTH_IMMUTABLE_USER_FIELDS``` which can be set to prevent overriding of fields and add them only if they are empty.
